velox/type/Type.h

```diff
@@ -2088,6 +2088,10 @@ struct CppToType<Timestamp> : public CppToTypeBase<TypeKind::TIMESTAMP> {};
 template <>
 struct CppToType<Date> : public CppToTypeBase<TypeKind::INTEGER> {};
 
+template <>
+struct CppToType<IntervalYearMonthType>
```

Comment:
Curious why is this needed.

Reply:
Got compilation error without this change:

```
/home/sparkuser/github/oap-project/velox/./velox/expression/UdfTypeResolver.h:78:9: error: no type named ‘NativeType’ in ‘struct facebook::velox::CppToType<facebook::velox::IntervalYearMonthType>’
   78 |   using in_type = typename CppToType<T>::NativeType;
      |         ^~~~~~~
/home/sparkuser/github/oap-project/velox/./velox/expression/UdfTypeResolver.h:80:9: error: no type named ‘NativeType’ in ‘struct facebook::velox::CppToType<facebook::velox::IntervalYearMonthType>’
   80 |   using out_type = typename CppToType<T>::NativeType;
```

Reply:
I see. The problem is that you are using IntervalYearMonthType instead of IntervalYearMonth (remove Type suffix). Once you fix that you can remove this code.

Reply:
See

```
template <>
struct resolver<IntervalYearMonth> {
  using in_type = int32_t;
  using null_free_in_type = in_type;
  using out_type = int32_t;
};
```